src/include/duckdb/parser/parsed_data/table_data_info.hpp

```diff
@@ -0,0 +1,164 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/parser/parsed_data/table_data_info.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/parser/parsed_data/parse_info.hpp"
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/common/vector.hpp"
+#include "duckdb/common/types/value.hpp"
+#include "duckdb/storage/table/column_data.hpp"
+
+namespace duckdb {
+
+struct TableDataInfo : public ParseInfo {
+public:
+	static constexpr const ParseInfoType TYPE = ParseInfoType::TABLE_DATA_INFO;
+
+public:
+	TableDataInfo() : ParseInfo(TYPE) {
+	}
+
+	//! Schema of the table
+	string schema;
+	//! Name of the table
+	string name;
+
+public:
+	unique_ptr<TableDataInfo> Copy() const {
+		auto result = make_uniq<TableDataInfo>();
+		result->schema = schema;
+		result->name = name;
+		return result;
+	}
+
+	void Serialize(Serializer &serializer) const override;
+	static unique_ptr<ParseInfo> Deserialize(Deserializer &deserializer);
+};
+
+struct InsertInfo : public TableDataInfo {
+public:
+	static constexpr const ParseInfoType TYPE = ParseInfoType::INSERT_INFO;
+
+public:
+	InsertInfo() : TableDataInfo() {
+	}
+
+	//! The chunk to insert
+	unique_ptr<DataChunk> chunk;
+
+public:
+	unique_ptr<InsertInfo> Copy() const {
+		auto result = make_uniq<InsertInfo>();
+		result->schema = schema;
+		result->name = name;
+		if (chunk) {
+			result->chunk = make_uniq<DataChunk>();
+			result->chunk->Copy(*chunk);
+		}
+		return result;
+	}
+
+	void Serialize(Serializer &serializer) const override;
+	static unique_ptr<ParseInfo> Deserialize(Deserializer &deserializer);
+};
+
+struct DeleteInfo : public TableDataInfo {
```

Comment:
This conflicts with the other [DeleteInfo](https://github.com/duckdb/duckdb/blob/main/src/include/duckdb/transaction/delete_info.hpp#L17)

Should I rename this one or namespace it?

Reply:
I'd rename this one, something like `TableDataDeleteInfo` is fine, we don't generally namespace things beyond `duckdb`, except for things in `third_party`

Reply:
Renamed